### PR TITLE
Update other-editors.scrbl to include langserver

### PR DESCRIPTION
**Pull request by spdegabrielle**
##### Checklist

- [ ] Bugfix
- [ ] Feature
- [ ] tests included
- [x] documentation

### Description of change
add Racket-langserver to https://docs.racket-lang.org/guide/other-editors.html

### CODE REVIEW

**Comment by spdegabrielle** [comment]
see also https://github.com/racket/racket/pull/4925

**Comment by beelzebielsk** [comment]
Two notes: I personally would not include capabilities of the language server in a discussion about setting up your editor. That's really a detail of the language server and I wouldn't want to update that every time the language server is capabilities change or grow. Second note is I don't believe bracket language server is under active development. If I remember correctly it is stalled and the maintainers have said so in an issue somewhere.

**Comment by spdegabrielle** [comment]
> Two notes: I personally would not include capabilities of the language server in a discussion about setting up your editor. That's really a detail of the language server and I wouldn't want to update that every time the language server is capabilities change or grow. Second note is I don't believe bracket language server is under active development. If I remember correctly it is stalled and the maintainers have said so in an issue somewhere.

Done

File: pkgs/racket-doc/scribblings/guide/other-editors.scrbl [hunk 1]
```diff
@@ -284,3 +284,39 @@ provides support for syntax highlighting and building for Sublime Text.
 
 The @hyperlink["https://marketplace.visualstudio.com/items?itemName=evzen-wybitul.magic-racket"]{Magic Racket}
 extension provides Racket support including REPL integration and syntax highlighting in Visual Studio Code.
+
+@section{racket-langserver}
+
+The @tt{racket-langserver} is a @hyperlink["http://langserver.org/"]{Language Server Protocol}
+implementation for Racket. This project seeks to use @hyperlink["https://github.com/racket/drracket"]{DrRacket's public APIs}
```
Comment (LiberalArtist) [inline pkgs/racket-doc/scribblings/guide/other-editors.scrbl:291]: If you want a link here, I think it should be `@seclink[#:indirect? #t #:doc '(lib "scribblings/drracket-tools/drracket-tools.scrbl") "top"]{DrRacket's public APIs}`.

File: pkgs/racket-doc/scribblings/guide/other-editors.scrbl [hunk 2]
```diff
@@ -284,3 +284,39 @@ provides support for syntax highlighting and building for Sublime Text.
 
 The @hyperlink["https://marketplace.visualstudio.com/items?itemName=evzen-wybitul.magic-racket"]{Magic Racket}
 extension provides Racket support including REPL integration and syntax highlighting in Visual Studio Code.
+
+@section{racket-langserver}
+
+The @tt{racket-langserver} is a @hyperlink["http://langserver.org/"]{Language Server Protocol}
```
Comment (LiberalArtist) [inline pkgs/racket-doc/scribblings/guide/other-editors.scrbl:290]: This should use `https`.

File: pkgs/racket-doc/scribblings/guide/other-editors.scrbl [hunk 3]
```diff
@@ -284,3 +284,39 @@ provides support for syntax highlighting and building for Sublime Text.
...
+@subsection{Installation and usage}
+
+A Racket runtime is a prerequisite, so before using @tt{racket-langserver}, ensure that a Racket runtime 
+is installed. You can install an from the @hyperlink["https://download.racket-lang.org"]{official download page}
+or install one from your package manager.
+
+@subsection{Atom}
+
+You can use the @hyperlink["https://github.com/cfinegan/atom-ide-racket"]{atom-ide-racket} package. The language 
+server will be automatically installed when @tt{atom-ide-racket} installs.
```
Comment (LiberalArtist) [inline pkgs/racket-doc/scribblings/guide/other-editors.scrbl:303]: Since Atom was discontinued over a year ago, I'd be inclined not to mention it here.

**Review by LiberalArtist — CHANGES_REQUESTED** [review]
I don't think The Racket Guide should provide all of these details. Instead, I think the first step should be to contribute Scribble documentation to https://github.com/jeapostrophe/racket-langserver (which may require a decision from its maintainers about how to structure the packaging to minimize dependencies). Then, I think the guide should get something like a `@section{Other Editors via the Language Server Protocol}` that indirect-links to the new `racket-langserver` documentation.

In particular, The Racket Guide is part of the Racket release cycle, and `racket-langserver` is not. Running `raco pkg install racket-langserver` will install the latest release of the package, not the version from the date of the Racket release, so it makes sense to link to documentation matching the package actually being installed.

**Comment by spdegabrielle** [comment]
> I don't think The Racket Guide should provide all of these details. Instead, I think the first step should be to contribute Scribble documentation to https://github.com/jeapostrophe/racket-langserver (which may require a decision from its maintainers about how to structure the packaging to minimize dependencies). Then, I think the guide should get something like a `@section{Other Editors via the Language Server Protocol}` that indirect-links to the new `racket-langserver` documentation.

an initial PR has been created at https://github.com/jeapostrophe/racket-langserver/pull/130

That said, I believe this is a pragmatic change that will enhance the experience for new Racket users.

File: pkgs/racket-doc/scribblings/guide/other-editors.scrbl [hunk 4]
```diff
@@ -284,3 +284,35 @@ provides support for syntax highlighting and building for Sublime Text.
...
+
+The @tt{racket-langserver} is a @hyperlink["https://langserver.org/"]{Language Server Protocol}
+implementation for Racket. This project seeks to use 
+@seclink[#:indirect? #t #:doc '(lib "scribblings/drracket-tools/drracket-tools.scrbl") "top"]{DrRacket's public APIs} 
+to provide functionality that mimics DrRacket's code tools as closely as possible.
+
+@subsection{Installation and usage}
+
+A Racket runtime is a prerequisite, so before using @tt{racket-langserver}, ensure that a Racket runtime 
+is installed. You can install an from the @hyperlink["https://download.racket-lang.org"]{official download page}
```
Comment (benknoble) [inline pkgs/racket-doc/scribblings/guide/other-editors.scrbl:298]: typo: `install an`

**Comment by mfelleisen** [comment]
Perhaps there should be a Newbie guide that includes such details.

File: pkgs/racket-doc/scribblings/guide/other-editors.scrbl (hunk 4 above)
Comment (spdegabrielle) [reply pkgs/racket-doc/scribblings/guide/other-editors.scrbl]: fixed

**Comment by spdegabrielle** [comment]
> Perhaps there should be a Newbie guide that includes such details.

@mfelleisen I see 'getting started' as the newbie guide.

#4925 adds a link to the editors chapter.

**Comment by spdegabrielle** [comment]
@mfelleisen is there any barrier to this PR ?
- if there is something wrong I can fix it.
-  if it is undesirable I will close it. 
 
I don't want it hanging around like a bad smell. 

I won't be creating an newbie guide today. if you like this PR can be considered an interim measure until a newbie guide is created.

**Pull request closed without merging**